app/renderer/components/main/main.js

```diff
@@ -402,12 +384,11 @@ class Main extends ImmutableComponent {
       windowActions.setDownloadsToolbarVisible(false)
     })
 
-    const self = this
     ipc.on(messages.BLOCKED_RESOURCE, (e, blockType, details) => {
-      const frameProps = frameStateUtil.getFrameByTabId(self.props.windowState, details.tabId)
-      frameProps && windowActions.setBlockedBy(frameProps, blockType, details.url)
+      windowActions.setBlockedBy(this.props.activeFrameKey, blockType, details.url)
     })
 
+    // TODO can we remove this?
     ipc.on(messages.BLOCKED_PAGE, (e, blockType, details) => {
```

Comment:
not sure what it would do if it blocked the whole page. We also have to be very careful about what we do inside `onBeforeRequest` because it blocks resource loading. Created https://github.com/brave/browser-laptop/pull/9702 to make it more efficient, but we should be saving the data in appState and reading the blocked resources from there